strabo/static/js/map.js

```diff
@@ -37,7 +37,24 @@ $(document).ready(function(){
     //set_map_click(map);
 
     set_geolocation(map);
+
+});
+
+$(document).keyup(function(e) {
+    if (e.keyCode==27) { //if ESC key is hit
+        hide_popup();
+        gallery.close();
+    }
 });
+
+$(document).mouseup(function(e){
+        var popup = $(".popup");
+        var popup_foreground = $(".popup-foreground");
+        if (popup.is(e.target) && popup.has(e.target).length===0 && !popup_foreground.is(e.target)){
```

Comment:
It looks like the !popup_foreground.is(e.target)
is not actually necessary, I deleted it and it works fine.

Also, to make this more consistent with my other events, you could put it in its own little function and do

    $(".popup").click(function(e){
          hide_popup();
     }

Also, this is all popup code, so maybe it should be moved to popup.js?